### ISSUE TITLE
Fails to start: with "_on_connect() missing 1 required positional argument: 'initial_data"

### ISSUE THREAD

**Issue by jonesn3** [xmikos/hangupsbot#8]
Hangups is global and works correctly. I recently ran this bot on an older computer without problem, maybe issues with a newer version of hangups?

Client unexpectedly disconnected:
_on_connect() missing 1 required positional argument: 'initial_data'
Traceback (most recent call last):
  File "/home/user/projects/hubot/node_modules/hubot-hangups/HangoutsBot/hangupsbot.py", line 137, in run
    loop.run_until_complete(self._client.connect())
  File "/usr/lib/python3.4/asyncio/base_events.py", line 208, in run_until_complete
    return future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 243, in result
    raise self._exception
  File "/usr/lib/python3.4/asyncio/tasks.py", line 298, in _step
    result = coro.throw(exc)
  File "/usr/local/lib/python3.4/dist-packages/hangups/client.py", line 99, in connect
    yield from self._listen_future
  File "/usr/lib/python3.4/asyncio/futures.py", line 348, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 351, in _wakeup
    value = future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 243, in result
    raise self._exception
  File "/usr/lib/python3.4/asyncio/tasks.py", line 300, in _step
    result = coro.send(value)
  File "/usr/local/lib/python3.4/dist-packages/hangups/channel.py", line 214, in listen
    yield from self._longpoll_request()
  File "/usr/local/lib/python3.4/dist-packages/hangups/channel.py", line 343, in _longpoll_request
    yield from self._on_push_data(chunk)
  File "/usr/local/lib/python3.4/dist-packages/hangups/channel.py", line 364, in _on_push_data
    yield from self.on_connect.fire()
  File "/usr/local/lib/python3.4/dist-packages/hangups/event.py", line 50, in fire
    yield from gen
  File "/usr/local/lib/python3.4/dist-packages/hangups/event.py", line 48, in fire
    gen = observer(_args, *_kwargs)
TypeError: _on_connect() missing 1 required positional argument: 'initial_data'

**Comment by luckymancvp** [comment]
me2

**Comment by wardellbagby** [comment]
@luckymancvp, and @BigJones, It'll probably be a while before @xmikos takes a look at this to get it fixed, but if I figure it out before he does, I'll submit a pull request to his branch so you all can use that. 

Really, I'd just recommend downgrading to back to hangups 0.2.10, since 0.3.0 is the one that causes this crash.

**Comment by wardellbagby** [comment]
@luckymancvp and @BigJones, I'm a man of my word and the pull request is now active. I've only done light testing but I can verify that it does indeed start-up and it will run. I haven't tested all of the commands, but I fixed all of the ones that I saw would be an issue (generally the ones related to changing things in the conversation.) 

This is, of course, based off of @xmikos and not my own, so there should be no incompatibility issues temporarily using my branch until either @xmikos accepts the pull request or denies it in lieu of his own. Check out #9 for the info on my changes.
